src/modules/api/accessControl/services/index.ts

```diff
@@ -0,0 +1,115 @@
+import { PrismaService } from "@/modules/core/prisma/services";
+import { Prisma } from "@prisma/client";
+import { CreateRoleDto, FetchRolesDto } from "../dtos";
+import { buildResponse } from "@/utils";
+import { HttpStatus, Injectable } from "@nestjs/common";
+import slugify from "slugify";
+import { DuplicateRoleException, RoleNotFoundException } from "../errors";
+
+@Injectable()
+export class AccessControlService {
+    constructor(private prisma: PrismaService) {}
+
+    async fetchRoles(options: FetchRolesDto) {
+        const queryOptions: Prisma.RoleFindManyArgs = {
+            where: {},
+            select: {
+                name: true,
+            },
+        };
+
+        if (options.roleName) {
+            queryOptions.where.name = { search: options.roleName };
+        }
+
+        const roles = await this.prisma.role.findMany(queryOptions);
+
+        return buildResponse({
+            message: "Roles retrieved successfully",
+            data: roles,
+        });
+    }
+
+    private async validatePermission(permission: Array<number>) {
+        const permissions = await this.prisma.permission.findMany({
+            where: {
+                id: {
+                    in: permission,
+                },
+            },
+        });
+
+        return permissions.length === permission.length;
+    }
+
+    async createRoles(options: CreateRoleDto) {
+        const roleExists = await this.prisma.role.findUnique({
+            where: {
+                name: options.roleName,
+            },
+        });
+        if (roleExists) {
+            throw new DuplicateRoleException(
+                "A role with this name already exists",
+                HttpStatus.BAD_REQUEST
+            );
+        }
+
+        const slug = slugify(options.roleName, { lower: true });
```

Comment:
Consider creating a utility function that wraps the slugify function so that it can also be used in any part of the codebase. Also set the strict option flag of the slugify function.